streaming/src/test/scala/org/apache/spark/streaming/StreamingContextSuite.scala

```diff
@@ -796,3 +821,21 @@ package object testPackage extends Assertions {
     }
   }
 }
+
+/**
+ * Helper methods for testing StreamingContextSuite.
+ * This includes methods to access private methods and fields in StreamingContext and MetricsSystem.
+ */
+private object StreamingContextSuite extends PrivateMethodTester {
+  private val _sources = PrivateMethod[ArrayBuffer[Source]]('sources)
+  private def getSources(metricsSystem: MetricsSystem): ArrayBuffer[Source] = {
+    metricsSystem invokePrivate _sources()
```

Comment:
You don't need postfix syntax here; `metricsSystem.invokePrivate(_sources())` is probably better.

Your test fails because the streaming source isn't present after the streaming context stops. Isn't that the point? is your test negated?